src/libNode/Node.cpp

```diff
@@ -1112,23 +1112,15 @@ bool Node::StartRetrieveHistory(const SyncType syncType,
         m_mediator.m_ds->m_shards, m_mediator.m_ds->m_publicKeyToshardIdMap,
         m_mediator.m_ds->m_mapNodeReputation);
   }
+  bool rejoinCondition = REJOIN_NODE_NOT_IN_NETWORK && !LOOKUP_NODE_MODE && !bDS;
+  LOG_GENERAL(INFO, "rejoinCondition = "<< rejoinCondition); //TODO: remove this log before merge
 
-  if (REJOIN_NODE_NOT_IN_NETWORK && !LOOKUP_NODE_MODE && !bDS) {
-    if (!bInShardStructure) {
-      LOG_GENERAL(
-          WARNING,
-          "Node " << m_mediator.m_selfKey.second
-                  << " is not in network, apply re-join process instead");
+  if (rejoinCondition && bIpChanged) {
+    LOG_GENERAL(
+        INFO, "My IP has been changed. So will broadcast my new IP to network");
+    if (!UpdateShardNodeIdentity()) {
```

Comment:
In theory it should be possible for a node's identity to change, but the IP address not to change. e.g. Kubernetes could just shuffle the same pods around different k8s nodes. Should we try to handle this case?

Reply:
If identity changed because of pods reshuffling, this would still be considered IP change from node per see.
So I think this should still work without any change.